src/network_helper_win.rs

```diff
@@ -66,13 +66,14 @@ pub(crate) fn get_interface_address() -> Result<InterfaceAddressIterator, String
         // https://learn.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersaddresses#remarks
         // A 15k buffer is recommended
         let mut size: u32 = 15 * 1024;
-        let mut buf = null_mut();
         let mut ret = ERROR_SUCCESS;
 
         // https://learn.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersaddresses#examples
         // Try to retrieve adapter information up to 3 times
         for _ in 0..3 {
-            buf = libc::malloc(size as usize) as PIP_ADAPTER_ADDRESSES;
+            let buf = libc::malloc(size as _) as PIP_ADAPTER_ADDRESSES;
+            // free memory on drop
+            let iterator = InterfaceAddressIterator { buf, adapter: buf };
```

Comment:
If you do that, it means `buf` will be freed and reallocated at every iteration. Unless that's what you want, not sure it's the best way forward.